File: sub-mysql-web-service/src/main/java/org/github/mysql/web/shiro/mapper/IShiroMapper.java
package org.github.mysql.web.shiro.mapper;

import lombok.NonNull;
import org.apache.ibatis.annotations.Param;
import org.github.mybatis.MyBatisMapper;
import org.github.mysql.web.shiro.model.UserDTO;
import org.github.shiro.AuthorInfo;

import javax.annotation.Nullable;
import java.util.List;

@MyBatisMapper
public interface IShiroMapper {
  @NonNull List<AuthorInfo> queryAuthorInfo(@NonNull @Param("userId") Long userId);

  @Nullable UserDTO queryUserInfo(@NonNull @Param("username") String username);
}

File: sub-mysql-mydb-service/src/main/java/org/github/mysql/mydb/auth/mapper/IShiroMapper.kt
package org.github.mysql.mydb.auth.mapper

import org.apache.ibatis.annotations.Param
import org.github.mybatis.MyBatisMapper
import org.github.mysql.mydb.model.auth.UserDTO
import org.github.shiro.AuthorInfo

@MyBatisMapper
interface IShiroMapper {
  fun queryAuthorInfo(@Param("userId") userId: Long): List<AuthorInfo>

  fun queryUserInfo(@Param("username") username: String): UserDTO?
}
